File: app/src/main/java/com/aa/androidormcomparison/measures/MeasuresConductor.java
package com.aa.androidormcomparison.measures;

import java.util.Arrays;
import java.util.List;

public class MeasuresConductor {
    final static List<Integer> NUMBERS_OF_ENTITIES = Arrays.asList(500, 1000, 1500, 2000, 2500);
    final static int NUMBER_OF_MEASURES = 1;
    Measurer measurer;

    public MeasuresConductor(Measurer measurer) {
        this.measurer = measurer;
    }

    public void conductMeasures() {
        for (int i = 0; i < NUMBER_OF_MEASURES; i++) {
            for (Integer numberOfEntities : NUMBERS_OF_ENTITIES) {
                measurer.conductWholeMeasureProcess(numberOfEntities);
            }
        }
    }
}

File: app/src/main/java/com/aa/androidormcomparison/measures/MeasuresConductor.kt
package com.aa.androidormcomparison.measures

import java.util.concurrent.Executors

class MeasuresConductor(val measurer: Measurer) {
    var NUMBERS_OF_ENTITIES = listOf(250)//5000, 10_000, 20_000)
    var NUMBER_OF_MEASURES = 1
    fun conductMeasures() {
        val executorService = Executors.newSingleThreadExecutor()
        executorService.execute {
            for (i in 1..NUMBER_OF_MEASURES) {
                for (numberOfEntities in NUMBERS_OF_ENTITIES) {
                    measurer.conductWholeMeasureProcess(numberOfEntities)
                }
            }
        }
    }
}